iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/AbstractEnvVariableMonitorCheck.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.checks;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+import org.sonar.iac.docker.tree.api.EnvInstruction;
+import org.sonar.iac.docker.tree.api.FromInstruction;
+import org.sonar.iac.docker.tree.api.KeyValuePair;
+
+public abstract class AbstractEnvVariableMonitorCheck implements IacCheck {
+  private final Map<String, String> envVariables = new HashMap<>();
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(EnvInstruction.class, this::updateEnvVariable);
+    init.register(FromInstruction.class, this::cleanEnvVariables);
+    init(init);
```

Comment:
This combination is a bit dangerous, as children classes may override the wrong init method.
I don't see a better approach though atm :/

Reply:
Same feeling, I'm not totally satisfied with this approach but I couldn't find better :sweat_smile: